src/user/UserCredentials.tsx

```diff
@@ -0,0 +1,493 @@
+  const [open, setOpen] = useState(false);
+  const [openSaveConfirm, setOpenSaveConfirm] = useState(false);
+  const [kebabOpen, setKebabOpen] = useState(false);
+  const adminClient = useAdminClient();
+  const form = useForm<CredentialsForm>({ defaultValues });
+  const { control, errors, handleSubmit, register } = form;
+  const [credentials, setCredentials] = useState<CredentialsForm>();
+  const [userCredentials, setUserCredentials] = useState<
+    CredentialRepresentation[]
+  >([]);
+  const [selectedCredential, setSelectedCredential] =
+    useState<CredentialRepresentation>({});
+  const [isResetPassword, setIsResetPassword] = useState(false);
+  const [showData, setShowData] = useState(false);
+
+  useFetch(
+    () => adminClient.users.getCredentials({ id: user.id! }),
+    (credentials) => {
+      setUserCredentials(credentials);
+    },
+    [key]
+  );
+
+  const passwordWatcher = useWatch({
+    control,
+    name: "password",
+  });
+
+  const passwordConfirmationWatcher = useWatch({
+    control,
+    name: "passwordConfirmation",
+  });
+
+  const isNotDisabled =
+    passwordWatcher !== "" && passwordConfirmationWatcher !== "";
+
+  const toggleModal = () => {
+    setOpen(!open);
+  };
+
+  const toggleConfirmSaveModal = () => {
+    setOpenSaveConfirm(!openSaveConfirm);
+  };
+
+  const saveUserPassword = async () => {
+    const passwordsMatch =
+      credentials?.password === credentials?.passwordConfirmation;
+
+    try {
+      await adminClient.users.resetPassword({
+        id: user.id!,
+        credential: {
+          temporary: credentials?.temporaryPassword,
+          type: "password",
+          value: passwordsMatch ? credentials?.password : "",
+        },
+      });
+      refresh();
+      addAlert(
+        isResetPassword
+          ? t("resetCredentialsSuccess")
+          : t("savePasswordSuccess"),
+        AlertVariant.success
+      );
+      setIsResetPassword(false);
+      setOpenSaveConfirm(false);
+    } catch (error) {
+      addError(
+        !passwordsMatch
+          ? isResetPassword
+            ? t("resetPasswordNotMatchError")
+            : t("savePasswordNotMatchError")
+          : isResetPassword
+          ? t("resetPasswordError")
+          : t("savePasswordError"),
+        error
+      );
+    }
+  };
+
+  const resetPassword = () => {
+    setIsResetPassword(true);
+    setOpen(true);
+  };
+
+  const [toggleDeleteDialog, DeleteConfirm] = useConfirmDialog({
+    titleKey: t("deleteCredentialsConfirmTitle"),
+    messageKey: t("deleteCredentialsConfirm"),
+    continueButtonLabel: t("common:delete"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
+      try {
+        await adminClient.users.deleteCredential({
+          id: user.id!,
+          credentialId: selectedCredential.id!,
+        });
+        addAlert(t("deleteCredentialsSuccess"), AlertVariant.success);
+        setKey(key + 1);
```

Comment:
This might hold an older version of the `key` due to the way scoping works in JS. Perhaps we can use the callback style here?
```suggestion
        setKey((key) => key + 1);
```